File: src/java/engine/movegen/bitboards/QueenMoveGenerator.java
package java.engine.movegen.bitboards;

import java.util.function.BiFunction;
import java.util.stream.Stream;

/**
 * Created by melvic on 7/30/18.
 */
public interface QueenMoveGenerator extends BishopMoveGenerator, RookMoveGenerator {
    public default Stream<BiFunction<Integer, Long, Long>> getAllMoves() {
        return Stream.concat(BishopMoveGenerator.super.getAllMoves(), RookMoveGenerator.super.getAllMoves());
    }
}

File: core/src/com/github/melvic/nostalgia/engine/movegen/bitboards/QueenMoveGenerator.scala
package com.github.melvic.nostalgia.engine.movegen.bitboards

/**
  * Created by melvic on 12/17/18.
  */
object QueenMoveGenerator extends SlidingMoveGenerator {
  override def moves = BishopMoveGenerator.moves ++ RookMoveGenerator.moves
}
